api/v1beta1/verticarestorepointsquery_types.go

```diff
@@ -23,21 +23,43 @@ import (
 
 // VerticaRestorePointsQuerySpec defines the desired state of VerticaRestorePointsQuery
 type VerticaRestorePointsQuerySpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
 	// +kubebuilder:validation:Required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
 	// The name of the VerticaDB CR that this VerticaRestorePointsQuery is defined for.  The
 	// VerticaDB object must exist in the same namespace as this object.
 	VerticaDBName string `json:"verticaDBName"`
 
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:advanced"
+	// Optional parameter that will limit the query to only restore points satisfying provided filter options
+	FilterOptions *VerticaRestorePointQueryFilterOptions `json:"filterOptions,omitempty"`
+}
+
+// VerticaRestorePointQueryFilterOptions defines the filter options to use while listing restore points
+type VerticaRestorePointQueryFilterOptions struct {
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// Optional parameter that will limit the query to only restore points from this archive
+	ArchiveName *string `json:"archiveName,omitempty"`
```

Comment:
Do these fields need to be pointers? We can just treat an empty string as if it isn't set.

Reply:
So either when the user explicitly set this to `""` or when user doesn't set this we always pass empty string to vclusterops and them NMA, which essentially ignores all empty string values for all filtering options?

Reply:
Yes, that was what I was thinking. But this may be moot if we use the struct directly from the vclusterops package.